tooling/metamodel-generator/src/main/java/org/hibernate/jpamodelgen/xml/XmlMetaEntity.java

```diff
@@ -339,7 +339,7 @@ private String getType(String propertyName, String explicitTargetEntity, Element
 					return "java.lang.Double";
 				}
 				case DECLARED: {
-					return mirror.toString();
+					return ((DeclaredType)mirror).asElement().asType().toString();
```

Comment:
So, `getQualifiedName()` won't work as it's based on `getCanonicalName()`.

The only option we have is to use `getEnclosingElement()` and concatenates the `getSimpleName()` until we find an element that is a `PackageElement` and then we can use the `getQualifiedName()` of the package.

Reply:
if i understood right I tried to use next expression
```java
((QualifiedNameable)((DeclaredType) mirror).asElement()).getQualifiedName().toString();
```
It's work fine for nested classes, but the `EmbeddableConfiguredInXmlTest` fails.
Fail this assertion
```java
assertTrue( entityMetaModel.contains( "SingularAttribute<BusinessEntity, BusinessId<T>> businessId" ) );
```
because source code contain `BusinessId` type without parameter
```java
public static volatile SingularAttribute<BusinessEntity, BusinessId> businessId;
```

Reply:
But I'm not sure, that this assertion is correct. Because generated line
```java
public static volatile SingularAttribute<BusinessEntity, BusinessId<T>> businessId;
```
not compiled (of course `cannot find symbol T`) and test not failed only thanks by `@IgnoreCompilationErrors` annotation.